### PR TITLE
Improve README and contributor documentation

### PR DESCRIPTION
**Pull request by Luthaf**
This contains improvements for the language and update the build instructions for Lumol.

- [New README](https://github.com/lumol-org/lumol/blob/fcbd01120e275f3985815f48223dc287af9f81bf/README.md)
- [New contributor instructions](https://github.com/lumol-org/lumol/blob/fcbd01120e275f3985815f48223dc287af9f81bf/Contributing.md)

It also change the goals of the framework to be **flexible**, **reliable** and **extensible**. I did not put fast and parallel in them, because I want to wait for #49 verdict before saying so; and because the code is not yet parallel (see #56 for the discussion about this).

Do you agree with this set of goal? Am I missing something?

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -1,17 +1,15 @@
-So you want to contribute to Lumol and create the next generation of
-molecular simulation engine? Great! This document tries to explain how you can
-contribute, but in case of doubt do not hesitate to contact me at
-luthaf@luthaf.fr.
+So you want to contribute to Lumol and create the next generation of molecular
+simulation engine? Great! This document explain how you can contribute, but in
```
Comment (g-bauer) [inline Contributing.md]: This document explains...

File: README.md [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
+Documentation is hosted [here](http://lumol-org.github.io/lumol), and separated
+in two parts:
+
+- The [user manual][user_manual] contains information on how to use Lumol as a
+  command line tool, and the complete input file documentation. Use this
+  documentation if you want to use Lumol as a simulation engine — without
+  writing code.
+- To use Lumol as a library inside your own code, we have a [developer
+  documentation][devdoc], which contains documentation for all the library
+  public functions, and example for most of them.
```
Comment (g-bauer) [inline README.md]: ...and examples...

File: Contributing.md (hunk 1 above)
Comment (Luthaf) [inline Contributing.md]: Thanks!

**Comment by g-bauer** [comment]
> flexible, reliable and extensible

Yes, I think these are nice goals.

**Pull request merged**